libs/article-parser/src/lib/correlation-score/correlation-score.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  ParsedArticleHead,
+  Parser,
+  ParsedArticle,
+  ParsedArticleParagraph,
+} from '@foodmedicine/interfaces';
+import * as fetch from 'node-fetch';
+import { correlationWeights, cutOffs } from './correlation-constants';
+import * as natural from 'natural';
+
+const tokenizer = new natural.WordTokenizer();
+
+async function downloadArticle(url: string): Promise<string> {
+  const ret = await fetch(url);
+  return await ret.text();
+}
+
+/**
+ * Find word frequencies through fuzzy search
+ * @param word
+ * @param paragraph
+ */
+function findWordFreqFuzzy(word: string, paragraph: string): number {
+  const tokenizedParagraph = tokenizer.tokenize(paragraph);
+  const overallFreqScore = tokenizedParagraph.reduce(
+    (freq: number, paragraphWord) => {
+      // distance ranges from 0 to 1. 1 being a perfect match
+      const distance = natural.JaroWinklerDistance(word, paragraphWord);
+      return freq + (distance > cutOffs.minimumWordDistance ? distance : 0);
+    },
+    0
+  );
+  return overallFreqScore;
+}
+
+/**
+ * Compute the correlation score based off of the inputs
+ * Current features include impact frequencies, recommendation frequencies, impact x recommendation
+ * Paragraph length
+ * @param impactFreq
+ * @param recommendationFreq
+ * @param paragraphWordCount
+ */
+function computeScore(
+  impactFreq: number,
+  recommendationFreq: number,
+  paragraphWordCount: number
+): number {
+  const impactScore = impactFreq * correlationWeights.impactWordFreq;
+  const recommendationScore =
+    recommendationFreq * correlationWeights.recommendationWordFreq;
+  const crossScore =
+    impactFreq *
+    recommendationFreq *
+    correlationWeights.impactCrossRecommendation;
+  const paragraphLengthScore =
+    paragraphWordCount * correlationWeights.paragraphLength;
+  // ensures that both impact and recommendation are seen in the same paragraph
+  return impactScore + recommendationScore + crossScore + paragraphLengthScore;
+}
+
+function stemString(input: string) {
+  return natural.PorterStemmer.tokenizeAndStem(input).join(' ');
+}
+
+async function getParagraphCorrelationScore(
+  paragraph: string,
+  impacted: string,
+  recommendation: string
+): Promise<ParsedArticleParagraph> {
```

Comment:
It doesn't look like this returns a promise?

Reply:
it was async earlier which automatically returns a prom, but it did not need to be. Removed async element